### PR TITLE
Add a custom semantics sample

### PR DESCRIPTION
**Pull request by goderbauer**
This change requires https://github.com/flutter/flutter/pull/12741 to be submitted first.

### CODE REVIEW

File: examples/catalog/lib/custom_semantics.dart [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+/// "Adjustable" to screen readers (e.g. TalkBack on Android and VoiceOver on
+/// iOS).
+///
+/// This allows screen reader users to swipe up/down (on iOS) or use the volume
+/// keys (on Android) to switch between the values in the dropdown menu.
+/// Depending on what the values in the dropdown menu are this can be a more
+/// intuitive way of switching values compared to exposing the content of the
+/// drop down menu as a screen overlay from which the user can select.
+///
+/// Users, that do not use a screen reader, will just see a regular dropdown
```
Comment (gspencergoog) [inline examples/catalog/lib/custom_semantics.dart]: "Users that do not use a screen reader will just see a regular dropdown"

**Review by gspencergoog — APPROVED** [review]

**Comment by gspencergoog** [comment]
lgtm

**Comment by Hixie** [comment]
Is this something we should just offer out of the box?

**Comment by goderbauer** [comment]
I am hesitant. This is not the default behavior for OEM drop downs on the platforms and it's not appropriate for all kinds of drop downs.

**Pull request merged**